src/pages/BrowseHapps/BrowseHapps.test.js

```diff
@@ -68,11 +70,24 @@ const mocks = [
   disableHappMock
 ]
 
+function renderWithRouter (
+  ui,
+  {
+    route = '/',
+    history = createMemoryHistory({ initialEntries: [route] })
+  } = {}
+) {
+  return {
+    ...render(<Router history={history}>{ui}</Router>),
+    history
+  }
+}
+
 describe('BrowseHapps', () => {
   it('renders', async () => {
     let getAllByRole
     await act(async () => {
-      ({ getAllByRole } = render(<MockedProvider mocks={mocks} addTypename={false}>
+      ({ getAllByRole } = renderWithRouter(<MockedProvider mocks={mocks} addTypename={false}>
```

Comment:
why is this necessary?

Reply:
Mainly so that Link is not throwing errors. Sometimes it also handles actual navigation...

Reply:
my preference would be to only bring it in when we know we need it, and to not pass in a dummy history object in those cases

Reply:
I agree, though sometimes you need dummy history anyways - Router itself doesn't inject the prop.

Reply:
ah, that's surprising to me

Reply:
it's because renderWithRouter supplies some generic in-memory router (so there is context for `Link`s to work with) but we're not using our whole router here.
Usually it's router that renders the component, while here (in tests) we're providing the component wrapped in mocked providers and what not. 😉